omc3/scripts/merge_kmod_results.py

```diff
@@ -0,0 +1,275 @@
+
+NAME = "NAME"  # Column containing the IP/Beam names
+BEAMS = ("B1", "B2")
+IPS = ("ip1", "ip5")
+
+HEADER_IMBALANCE = "LUMINOSITY_IMBALANCE"
+HEADER_REL_ERROR = "RELATIVE_ERROR"
+HEADER_EFF_BETA_IP1 = "EFFECTIVE_BETA_IP1"
+HEADER_REL_ERROR_IP1 = "RELATIVE_ERROR_IP1"
+HEADER_EFF_BETA_IP5 = "EFFECTIVE_BETA_IP5"
+HEADER_REL_ERROR_IP5 = "RELATIVE_ERROR_IP5"
+
+BPMWK = "BPMWK."  # elements starting with this are dropped
+
+
+def get_params():
+    params = EntryPointParameters()
+    params.add_parameter(
+        name="kmod_dirs",
+        type=pathlib.Path,
+        nargs="+",
+        required=True,
+        help=f"Path to kmod directories with stored KMOD measurement files,"
+             f"in particular {LSA_RESULTS}{EXT}",
+    )
+    params.add_parameter(
+        name="outputdir",
+        type=pathlib.Path,
+        help="Output directory where to write the result tfs",
+    )
+    return params
+
+
+@entrypoint(get_params(), strict=True)
+def merge_kmod_results(opt) -> tfs.TfsDataFrame:
+    """ Main function to merge K-Mod output.
+    See :mod:`omc3.scripts.merge_kmod_results`.
+    """
+    if opt.outputdir:
+        save_config(opt.outputdir, opt, script=__file__)  # creates outputdir
+
+    # Get the directories we need where the tfs are stored
+    ip_dir_names: List[pathlib.Path] = get_ip_dir_names(opt.kmod_dirs)
+
+    # Combine the lsa data
+    lsa_tfs = merge_tfs(ip_dir_names, f"{LSA_RESULTS}{EXT}")
+
+    # If the TFS data contains everything we need: get the imbalance
+    if _validate_for_imbalance(lsa_tfs):
+        imbalance_results = get_lumi_imbalance(lsa_tfs)
+        lsa_tfs = _add_imbalance_to_header(lsa_tfs, *imbalance_results)
+
+    if opt.outputdir:
+        output_path = opt.outputdir / f"{LSA_RESULTS}{EXT}"
+        LOG.info(f"Writing result TFS file to disk at: {str(output_path)}")
+        tfs.write(output_path, lsa_tfs, save_index=NAME)
+    return lsa_tfs
+
+
+def _check_tfs_sanity(data_frame: tfs.TfsDataFrame):
+    """
+    Checks that the merged `TfsDataFrame` has valid entries,
+    i.e. names are unique and max two entries per IP.
+
+    Args:
+        data_frame (tfs.TfsDataFrame): a loaded `TfsDataFrame` to validate.
+    """
+    # Check that both beams are there only once
+    multiple_names = [name for name in data_frame.index if (data_frame.index == name).sum() > 1]
+    if multiple_names:
+        msg = (f"Found entries '{', '.join(set(multiple_names))}' "
+               f"several times in merged DataFrame. "
+               "Expected only once")
+        LOG.error(msg)
+        raise KeyError(msg)
+
+    # check that there is no weird additional data
+    too_many_entries = [ip for ip in IPS if data_frame.index.str.startswith(ip).sum() > 2]
+    if too_many_entries:
+        msg = ("More than two entries found for ips "
+               f"{', '.join(too_many_entries)} in merged DataFrame. "
+               "Expected one for each beam.")
+        LOG.error(msg)
+        raise KeyError(msg)
```

Comment:
is this really needed? Can we have more than two unique indices and care about it at the same time?